http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -495,76 +480,103 @@ class ClusterMonitorActor(
+                                            runtimeStatus: RuntimeStatus
+                                          )(implicit ev: ApplicativeAsk[IO, TraceId]): IO[ClusterMonitorMessage]  = for {
+    dataprocInstances <- runtimeAndRuntimeConfig.runtimeConfig.cloudService match {
+      case CloudService.GCE      => IO.pure(Set.empty[DataprocInstance])
+      case CloudService.Dataproc => getDataprocInstances(runtimeAndRuntimeConfig.runtime)
+    }
+
+    runningInstanceCount = dataprocInstances.count(_.status == GceInstanceStatus.Running)
+    stoppedInstanceCount = dataprocInstances.count(
+      i => i.status == GceInstanceStatus.Stopped || i.status == GceInstanceStatus.Terminated
+    )
 
-      result <- googleStatus match {
-        case Unknown | Creating | Updating | Stopping | Starting =>
-          IO.pure(NotReadyCluster(runtimeAndRuntimeConfig, googleStatus, dataprocInstances))
-        // Take care we don't restart a Deleting or Stopping cluster if google hasn't updated their status yet
-        case Running
-            if runtimeAndRuntimeConfig.runtime.status != Deleting && runtimeAndRuntimeConfig.runtime.status != Stopping && runningInstanceCount == dataprocInstances.size =>
-          getMasterIp(runtimeAndRuntimeConfig).flatMap {
-            case Some(ip) => checkClusterTools(runtimeAndRuntimeConfig, ip, dataprocInstances)
-            case None =>
-              IO.pure(
-                NotReadyCluster(runtimeAndRuntimeConfig,
-                                RuntimeStatus.Running,
-                                dataprocInstances,
-                                Some("Could not retrieve master IP"))
+    result <- runtimeStatus match {
+      case Unknown | Creating | Updating | Stopping | Starting =>
+        IO.pure(NotReadyCluster(runtimeAndRuntimeConfig, runtimeStatus, dataprocInstances))
+      // Take care we don't restart a Deleting or Stopping cluster if google hasn't updated their status yet
+      case Running
+        if runtimeAndRuntimeConfig.runtime.status != Deleting && runtimeAndRuntimeConfig.runtime.status != Stopping && runningInstanceCount == dataprocInstances.size =>
+        getMasterIp(runtimeAndRuntimeConfig).flatMap {
+          case Some(ip) => checkClusterTools(runtimeAndRuntimeConfig, ip, dataprocInstances)
+          case None =>
+            IO.pure(
+              NotReadyCluster(runtimeAndRuntimeConfig,
+                RuntimeStatus.Running,
+                dataprocInstances,
+                Some("Could not retrieve master IP"))
+            )
+        }
+      // Take care we don't fail a Deleting or Stopping cluster if google hasn't updated their status yet
+      case Error
+        if runtimeAndRuntimeConfig.runtime.status != Deleting && runtimeAndRuntimeConfig.runtime.status != Stopping =>
+        runtimeAndRuntimeConfig.runtimeConfig.cloudService match {
+          // TODO implement getErrorDetails in wb-libs
```

Comment:
Now I remember running into this -- there's no such thing as `Error` status in GCE:
https://cloud.google.com/compute/docs/reference/rest/v1/instances/get
```
enum

[Output Only] The status of the instance. One of the following values: PROVISIONING, STAGING, RUNNING, STOPPING, STOPPED, SUSPENDING, SUSPENDED, and TERMINATED.
```

So this is a Dataproc-only concept..

Reply:
yea, but we're mapping gce statuses to `RuntimeStatus`, and deciding what to do for both gce and dataproc...It's not obvious what would be cleaner since branching out depends on gce and dataproc means we'll need to determine what to do based on gce or dataproc specific statuses as well

Reply:
now that I think about it..Maybe a better way to address this is to have different monitoring code for gce vs dataproc....since we also want different polling interval for them as well